woocommerce/woocommerce
# Add/crud get total count selector

### All Submissions:






### Changes proposed in this Pull Request:

Add's a getItemsTotalCount selector to the crud data store, so we can use it for the product categories data store.
This is required for the category dropdown field.

### How to test the changes in this Pull Request:

1. Load this branch, finish the onboarding and create some Product categories ( an easy way is also by importing the sample products in the **Personalize your store** task, which adds categories by default.
2. Go to **WooCommerce > Home** and open your console.
3. Retrieve a normal list of product categories `wp.data.select('experimental/wc/admin/products/categories').getProductCategories({ per_page: 5 })`
4. Test the total count selector: `wp.data.select('experimental/wc/admin/products/categories').getProductCategoriesTotalCount({ per_page: 5 })` this should return the total count, without triggering another request.
5. Test the total count selector: `wp.data.select('experimental/wc/admin/products/categories').getProductCategoriesTotalCount({ parent: 1 })` this should return `undefined` first and then the total count a second time, as it does trigger a new request.
6. Run `wp.data.select('experimental/wc/admin/products/categories').getProductCategories({ per_page: 5 })` one more time and look at the category list, and get one of the category id's
7. Run `wp.data.select('experimental/wc/admin/products/categories').getProductCategory( 222 )`, replacing 222 with the category id. It should return the category without triggering another request.
If you refresh the page and only run this, it should trigger a request and return the category when calling it a second time.


### Other information:




### FOR PR REVIEWER ONLY:
